### PR TITLE
Scripts

### PR DESCRIPTION
**Pull request by AlexSP0**
fix basescriptswidget, scriptswidget,  scriptspowershellwidget, scriptscommonslots, modelbuilder, scriptsmodelio

### CODE REVIEW

File: src/plugins/scripts/i18n/scripts_plugin_ru.ts [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        <source>User level policies.</source>
+        <translation>Политика для пользователя</translation>
+    </message>
+    <message>
+        <source>Logon</source>
+        <translation>При входе в систему</translation>
+    </message>
+    <message>
+        <source>Logoff</source>
+        <translation>При выходе из системы</translation>
```
Comment (august-alt) [inline src/plugins/scripts/i18n/scripts_plugin_ru.ts:79]: Change Russian translation to: "Выход из системы"

File: src/plugins/scripts/i18n/scripts_plugin_ru.ts [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    </message>
+    <message>
+        <source>Startup</source>
+        <translation>При загрузке компьютера</translation>
```
Comment (august-alt) [inline src/plugins/scripts/i18n/scripts_plugin_ru.ts:83]: Change Russian translation to: "Запуск"

File: src/plugins/scripts/i18n/scripts_plugin_ru.ts [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+    </message>
+    <message>
+        <source>Shutdown</source>
+        <translation>При выключении компьютера</translation>
```
Comment (august-alt) [inline src/plugins/scripts/i18n/scripts_plugin_ru.ts:87]: Change Russian translation to: "Завершение работы"

File: src/plugins/scripts/scriptitem.h [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+{
+public:
+    enum PropertyType
+    {
+        PATH      = 0,
+        PARAMETER = 1
+    };
+
+public:
+    //static inline const std::string PATH      = "path";
```
Comment (august-alt) [inline src/plugins/scripts/scriptitem.h:38]: Remove comments.

**Review by august-alt — CHANGES_REQUESTED** [review]

File: src/plugins/scripts/scriptstreemodel.cpp [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+    auto userSystemSetting = insertItem<ScriptsFolderItem>(userNamespace);
+    userSystemSetting->setDisplayName(QObject::tr("System settings").toStdString());
+    userSystemSetting->setProperty(ScriptsFolderItem::PARENT_ID, userUuid);
+    userSystemSetting->setProperty(ScriptsFolderItem::HELP_MSG, "System settings for user");
+
+    auto userScripts = insertItem<ScriptsFolderItem>(userSystemSetting);
+    userScripts->setDisplayName(QObject::tr("Scripts").toStdString());
+    userScripts->setProperty(ScriptsFolderItem::NAMESPACE, "User");
+    userScripts->setProperty(ScriptsFolderItem::HELP_MSG, "Scripts for user");
```
Comment (august-alt) [inline src/plugins/scripts/scriptstreemodel.cpp:111]: Help message is not translated.

File: src/plugins/scripts/scriptstreemodel.cpp [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    auto userUuid = QUuid("{123e4567-e89b-12d3-a456-426652340001}");
+    userNamespace->setProperty(ScriptsFolderItem::NODE_ID, userUuid);
+    userNamespace->setProperty(ScriptsFolderItem::PARENT_ID, topUuid);
+    userNamespace->setProperty(ScriptsFolderItem::HELP_MSG,
+                               QObject::tr("User level policies.").toStdString());
+
+    auto userSystemSetting = insertItem<ScriptsFolderItem>(userNamespace);
+    userSystemSetting->setDisplayName(QObject::tr("System settings").toStdString());
+    userSystemSetting->setProperty(ScriptsFolderItem::PARENT_ID, userUuid);
+    userSystemSetting->setProperty(ScriptsFolderItem::HELP_MSG, "System settings for user");
```
Comment (august-alt) [inline src/plugins/scripts/scriptstreemodel.cpp:106]: Help message is not translated.

File: src/plugins/scripts/scriptstreemodel.cpp [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+    auto machineSystemSettings = insertItem<ScriptsFolderItem>(machineNamespace);
+    machineSystemSettings->setDisplayName(QObject::tr("System settings").toStdString());
+    machineSystemSettings->setProperty(ScriptsFolderItem::PARENT_ID, machineUuid);
+    machineSystemSettings->setProperty(ScriptsFolderItem::HELP_MSG, "System settings for computer");
+
+    auto machineScripts = insertItem<ScriptsFolderItem>(machineSystemSettings);
+    machineScripts->setDisplayName(QObject::tr("Scripts").toStdString());
+    machineScripts->setProperty(ScriptsFolderItem::NAMESPACE, "Machine");
+    machineScripts->setProperty(ScriptsFolderItem::HELP_MSG, "Scripts for computer");
```
Comment (august-alt) [inline src/plugins/scripts/scriptstreemodel.cpp:92]: Help message is not translated.

File: src/plugins/scripts/scriptstreemodel.cpp [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+    auto machineUuid = QUuid("{123e4567-e89b-12d3-a456-426652340000}");
+    machineNamespace->setProperty(ScriptsFolderItem::NODE_ID, machineUuid);
+    machineNamespace->setProperty(ScriptsFolderItem::PARENT_ID, topUuid);
+    machineNamespace->setProperty(ScriptsFolderItem::HELP_MSG,
+                                  QObject::tr("Machine level policies").toStdString());
+
+    auto machineSystemSettings = insertItem<ScriptsFolderItem>(machineNamespace);
+    machineSystemSettings->setDisplayName(QObject::tr("System settings").toStdString());
+    machineSystemSettings->setProperty(ScriptsFolderItem::PARENT_ID, machineUuid);
+    machineSystemSettings->setProperty(ScriptsFolderItem::HELP_MSG, "System settings for computer");
```
Comment (august-alt) [inline src/plugins/scripts/scriptstreemodel.cpp:87]: Help message is not translated.

**Review by august-alt — CHANGES_REQUESTED** [review]

File: src/plugins/scripts/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+set(TS_FILES
+    ${CMAKE_CURRENT_SOURCE_DIR}/i18n/scripts_plugin_ru.ts
+    ${CMAKE_CURRENT_SOURCE_DIR}/i18n/scripts_plugin_en.ts
+)
+
+include_directories(${CMAKE_SOURCE_DIR}/src)
+
+set(PLUGIN_NAME scripts-plugin)
+
+#file(GLOB_RECURSE TS_FILES ${CMAKE_CURRENT_SOURCE_DIR} *.ts)
```
[the remark this answers is not in the document]
Comment (AlexSP0) [reply src/plugins/scripts/CMakeLists.txt]: fixed

File: src/plugins/scripts/basescriptswidget.cpp [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+
+#include "../../gui/filedialogutils.h"
+
+#include <mvvm/model/modelutils.h>
+#include <mvvm/viewmodel/viewitem.h>
+#include <QDebug>
+
+namespace scripts_plugin
+{
+BaseScriptWidget::BaseScriptWidget(QWidget *p)
```
[the remark this answers is not in the document]
Comment (AlexSP0) [reply src/plugins/scripts/basescriptswidget.cpp]: class renamed, inheritance already implemented

File: src/plugins/scripts/scriptstreemodel.cpp (hunk 5 above)
Comment (AlexSP0) [reply src/plugins/scripts/scriptstreemodel.cpp]: fixed

File: src/plugins/scripts/scriptstreemodel.cpp (hunk 6 above)
Comment (AlexSP0) [reply src/plugins/scripts/scriptstreemodel.cpp]: fixed

File: src/plugins/scripts/scriptstreemodel.cpp (hunk 7 above)
Comment (AlexSP0) [reply src/plugins/scripts/scriptstreemodel.cpp]: fixed

File: src/plugins/scripts/scriptstreemodel.cpp (hunk 8 above)
Comment (AlexSP0) [reply src/plugins/scripts/scriptstreemodel.cpp]: fixed

**Review by august-alt — CHANGES_REQUESTED** [review]
Refactoring of *BaseScriptTabWidget::findRootItem required. We need to provide additional fields in tree model, to simplify widget and model.

**Comment by august-alt** [comment]
Additionally it would be preferable to cherry pick set of commits in separate branch instead of merging this whole branch.

**Pull request merged**